Update jammy image

rbradford opened the pull request:
* scripts: Add a script to automate the custom image construction
* docs: Point at custom image build script in documentation
* tests: Update version of Jammy image in use
* tests: Disable test_tpm test
* tests: Give test_vfio_user more RAM

rbradford committed:
scripts: Add a script to automate the custom image construction

Only for x86-64 right now but does include support for custom VFIO
image.

Signed-off-by: Rob Bradford <rbradford@rivosinc.com>

rbradford committed:
docs: Point at custom image build script in documentation

Remove manual steps and replace with a script.

Signed-off-by: Rob Bradford <rbradford@rivosinc.com>

rbradford committed:
tests: Update version of Jammy image in use

This version is generated with the new script and adds kexec-tools.

Fixes: #6726

Signed-off-by: Rob Bradford <rbradford@rivosinc.com>

TimePrinciple commented on scripts/build-custom-image.sh:
```diff
@@ -0,0 +1,35 @@
+sudo kpartx -d $IMAGE_NAME_BASE.raw
+cp $IMAGE_NAME_BASE.raw $IMAGE_NAME_BASE-custom-$(date "+%Y%m%d")-0.raw
+qemu-img convert -p -f raw -O qcow2 $IMAGE_NAME_BASE-custom-$(date "+%Y%m%d")-0.raw $IMAGE_NAME_BASE-custom-$(date "+%Y%m%d")-0.qcow2
+popd
```
```suggestion
popd || exit
```

ditto :)

TimePrinciple reviewed (commented):
Some suggestions :)

rbradford committed:
tests: Give test_vfio_user more RAM

When running with the newer jammy OS image error messages from Rust
Hypervisor Firmware about lack of ram when rebooting where observed.
Increase the RAM allocation to allow it to boot after a reboot.

Signed-off-by: Rob Bradford <rbradford@rivosinc.com>

rbradford committed:
tests: Adjust test_tpm memory usage

After updating the OS image to the latest jammy version this test now
requires more memory to run

Fixes: #6782

Signed-off-by: Rob Bradford <rbradford@rivosinc.com>

jinankjain commented:
@smit-gardhariya Please review if anything is missing for kexec test case.

TimePrinciple commented:
I think `SC2048` should be `SC2046`, a typo :(

My apologies!

rbradford committed:
tests: Limit number of test thread on aarch64

This system has a lot of cores (80) resulting in all the tests being
spawned simultaneously and leading to exhaustion of the available
memory. Instead limit the number of threads.

Signed-off-by: Rob Bradford <rbradford@rivosinc.com>

up2wing commented on scripts/build-custom-image.sh:
```diff
@@ -0,0 +1,38 @@
+mkdir -p mnt
+export ROOTFS=/dev/mapper/$(sudo kpartx -v -a $IMAGE_NAME_BASE.raw | grep "p1 " | cut -f 3 -d " ")
+sudo mount $ROOTFS mnt
+sudo mv mnt/etc/resolv.conf mnt/etc/resolv.conf.backup
```
Maybe adding a blank line here would make the script seems clearer.

up2wing commented on scripts/build-custom-image.sh:
```diff
@@ -0,0 +1,38 @@
+umount /proc
+history -c
+exit
+EOF
```
ditto.

russell-islam commented on scripts/build-custom-image.sh:
```diff
@@ -0,0 +1,38 @@
+sudo mv mnt/etc/resolv.conf mnt/etc/resolv.conf.backup
+cat >script <<EOF
+#!/bin/bash
+set -x
```
why not setting -e here?
